src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -908,14 +909,27 @@ int runShellCmd(const std::string& cmd, std::string& output)
 int xcldev::device::runTestCase(const std::string& exe,
     const std::string& xclbin, std::string& output)
 {
-    std::string testCasePath = dsaPath +
+    std::string dirPath;
+    struct stat st;
+
+    output.clear();
+    
+    if (stat(xsaPath.c_str(), &st) != 0) {
```

Comment:
While going forward in 2019.2, we will be releasing all of our deployment platforms in the /opt/xilinx/xsa directory.  For some customers, they may still have deployment packages from an earlier time.   

_This means that both the /opt/xilinx/dsa and /opt/xilinx/xsa will need to co-exist._  

This code seems to prevent examining the /opt/xilinx/dsa directories once a 2019.2 deployment platform (e.g., xsabin) is installed.  

Was that the intent?

Reply:
Talked with Lizhi, this is for xbutil validate, which will try to load test xclbin from /opt/xilinx/xsa or /opt/xilinx/dsa. Once user installed /opt/xilinx/xsa newer version, the tests in the xsa directory should be valid. Users don't have the ability to chose default/selected directory by using xbutil.

Reply:
Chatted with Steve, more thoughts needed here because we might have both dsa and xsa installed, but the xbutil will search for /opt/xilinx/xxx/firmware name/test, if the firware name is in dsa rather than xsa, we will fail to find tests. Will fix this.

Reply:
fixed